src/libcore/iter/iterator.rs

```diff
@@ -1514,6 +1514,24 @@ pub trait Iterator {
         false
     }
 
+    /// Completly consumes an iterator. This is a conveniant method for
+    /// functional programming with iterators.
+    ///
+    /// ```
+    /// let xs = [0, 1, 2, 3];
+    /// let mut sum = 0;
+    ///
+    /// xs.iter().map(|x| sum += x).drain();
```

Comment:
This is a more complex way to write `xs.iter().sum()` or even `xs.iter().for_each(|x| sum += x)`.